nixos/modules/programs/vim.nix

```
@@ -4,22 +4,20 @@ let
   cfg = config.programs.vim;
 in {
   options.programs.vim = {
-    defaultEditor = lib.mkOption {
-      type = lib.types.bool;
-      default = false;
-      description = ''
-        When enabled, installs vim and configures vim to be the default editor
-        using the EDITOR environment variable.
-      '';
-    };
+    enable = lib.mkEnableOption "Vi IMproved, an advanced text";
+
+    defaultEditor = lib.mkEnableOption "vim as the default editor";
 
     package = lib.mkPackageOption pkgs "vim" {
       example = "vim-full";
     };
   };
 
-  config = lib.mkIf cfg.defaultEditor {
-    environment.systemPackages = [ cfg.package ];
-    environment.variables = { EDITOR = lib.mkOverride 900 "vim"; };
+  config = lib.mkIf cfg.enable {
```

```suggestion
  # TODO: convert it into assert
  config = lib.mkIf (cfg.enable || cfg.defaultEditor) (lib.warnIf (cfg.defaultEditor && !cfg.enable) "programs.vim.defaultEditor will only work if programs.vim.enable is enabled" {
```

Not very pleasant, but we should at least make sure the user isn't thrown into a "where is my editor???" situation. I'm not sure how will formatting go here

---

Yeah makes sense did not think about that, added a slightly modified version of this.

---

Switched over to using `config.warnings` as `lib.warnIf`gave an infinity recursive error.

---

> Switched over to using `config.warnings` as `lib.warnIf`gave an infinity recursive error.

You're right, I forgot how to do this.